servers/lib/server/handlers/api/gitlab/user.coffee

```diff
@@ -0,0 +1,150 @@
+hat = require 'hat'
+async = require 'async'
+
+apiErrors = require '../errors'
+{ handleUsername } = require '../helpers'
+
+GenericHandler = require './generichandler'
+
+
+module.exports = class User extends GenericHandler
+
+  # event user_create
+  @create = (data, callback = -> ) ->
+
+    # validating req params
+    { error, username, email, firstName,
+      lastName, suggestedUsername } = @validateDataFor 'create', data
+
+    return callback error  if error
+
+    bongo     = @getBongo()
+    { JUser } = @getModels()
+
+    client = null
+
+    queue  = [
+
+      (next) ->
+        JUser.one { username }, (err, user) ->
+          if err or user
+            return next err ? { message: 'user already exists' }
+          next()
+
+      (next) ->
+        handleUsername username, suggestedUsername, (err, _username) ->
+          return next err  if err
+          username = _username
+          next()
+
+      (next) ->
+        context = { group: 'gitlab' }
+        bongo.fetchClient 1, context, (client_) ->
```

Comment:
I don't want to include `req` into this level, tried to create an abstract api which only does the operation itself, not relies on anything. I can make `fetchClient` to act based on parameters or I can make the `1` as `false` and handle the flow in that way. But I don't want to carry `req` up to here, think this as a 1st level of api for user creation, I'll move this out to the first level api implementation package (once we've something like that)